core/embed/extmod/modtrezorcrypto/modtrezorcrypto-aesgcm.h

```diff
@@ -0,0 +1,205 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "py/objstr.h"
+
+#include "aes/aesgcm.h"
+#include "memzero.h"
+
+/// package: trezorcrypto.__init__
+
+/// class aesgcm:
+///     """
+///     AES-GCM context.
+///     """
+typedef struct _mp_obj_AesGcm_t {
+  mp_obj_base_t base;
+  gcm_ctx ctx;
+  enum {
+    STATE_INIT,
+    STATE_ENCRYPTING,
+    STATE_DECRYPTING,
+    STATE_FINISHED,
+  } state;
```

Comment:
Would it make sense to introduce `STATE_FAIL` to which the context would fall when some its method raises an exception?

Reply:
I don't think it adds much value, but it doesn't hurt as a failsafe. https://github.com/trezor/trezor-firmware/pull/3625/commits/c9f043bd966411cdb0597b0f221ddefc46a5fc9d